sensor/common/detector/metrics/metrics.go

```diff
@@ -165,6 +165,14 @@ var (
 		Help:      "A counter that tracks the size of the detection network flow buffer",
 	}, []string{"Operation"})
 
+	// DetectorDeploymentBufferSize keeps track of the size of the detection deployment buffer.
+	DetectorDeploymentBufferSize = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "detector_deployment_buffer_size",
+		Help:      "A counter that tracks the size of the deployment buffer",
```

Comment:
I know that we use this pattern of naming for other queues, but I consider it confusing.

What we are recording here are the queue operations, so the following would be more appropriate:

```suggestion
		Name:      "detector_deployment_queue_operations_total",
		Help:      "A counter that tracks number of ADD and REMOVE operations on the deployment buffer queue. Current size of the queue can be calculated by subtracting the number of remove operations form the add operations.",
```

I would have 2 general suggestions regarding the metrics for the queues (globally):
1. Apply the naming scheme from this comment `<thing>_queue_operations_total`
2. Instead of having two metrics (1: add&remove, 2: drop), have one counter vector with 3 types of operations: add, remove, drop (or using queue terminology: push, pop, drop).

Would be cool to have a followup PR (for after 4.7) with such unification.
(that makes this comment optional for this PR, but let's have the follow-up ticket created if that is okay).